nengo/params.py

```diff
@@ -187,18 +233,37 @@ def __set__(self, instance, ndarray):
             ndarray = self.validate(instance, ndarray)
         self.data[instance] = ndarray
 
-    def validate(self, instance, ndarray):
-        ndim = len(self.shape)
-        try:
-            ndarray = np.asarray(ndarray, dtype=np.float64)
-        except TypeError:
-            raise ValueError("Must be a float NumPy array (got type '%s')"
-                             % ndarray.__class__.__name__)
-
-        if ndarray.ndim != ndim:
+    def validate(self, instance, ndarray):  # noqa: C901
+        if isinstance(ndarray, np.ndarray):
+            ndarray = ndarray.view()
```

Comment:
It seems strange to me to make the requirement for float, and specifically float64, at the parameter level. I would prefer to just keep whatever users give us. This can be especially useful for large data structures that we want to store but not necessarily cast to float64. You're right, though, that this means that backends have to be more careful about how they use the data in these parameters. Maybe this should be a separate change? I'm trying to remember why I changed it in this PR specifically.